File: CivitasRuby/lib/casilla_sorpresa.rb
# encoding:utf-8
require_relative 'casilla'
module Civitas
  class CasillaSorpresa < Casilla
    def initialize(nombre,mazo)
      super(nombre)
      @mazo = mazo
    end
    
    def recibe_jugador(iactual,todos)
      if jugador_correcto(iactual,todos)
        super
        sorpresa = @mazo.siguiente
        sorpresa.aplicar_a_jugador(iactual,todos)
      end
    end
  end
end

File: CivitasJava/src/Civitas/CasillaSorpresa.java
package Civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla {
    private MazoSorpresas mazo;
    
    CasillaSorpresa(String nombre, MazoSorpresas mazo){
        super(nombre);
        this.mazo = mazo;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {
        Sorpresa sorpresa;
        if(super.jugadorCorrecto(iactual, todos)){
            sorpresa = this.mazo.siguiente();
            super.informe(iactual, todos);
            sorpresa.aplicarAJugador(iactual, todos);
        }
    }
}
